Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -4091,6 +4091,9 @@ - (void)findInteraction:(UIFindInteraction *)interaction didEndFindSession:(UIFi
 {
     if ([self _searchableObject] == _contentView.get())
         [_contentView didEndTextSearchOperation];
+
+    if ([session isKindOfClass:[UITextSearchingFindSession class]])
+        [(UITextSearchingFindSession *)session setSearchableObject:[self _searchableObject]];
```

Comment:
Is there a way we can fix this without using SPI? It also seems strange that this logic runs when the find session "ends".

I am wondering if there is something we can call to trigger `findInteraction:sessionForView:` to recreate the session entirely.

Reply:
> Is there a way we can fix this without using SPI?

I don't believe so; I looked through the API for various alternatives but couldn't find any. 

> It also seems strange that this logic runs when the find session "ends".

Ya, I know. It's because this is also called when the session changes. ("Called when the current search session has changed or ended")

> I am wondering if there is something we can call to trigger findInteraction:sessionForView: to recreate the session entirely.

Not that I know of, since the docs say it is "Called when a find session is requested to begin by the user"

Reply:
Richard and I confirmed that this approach is necessary due to a limitation of the existing UIKit API.